pennylane/fourier/spectrum.py

```diff
@@ -0,0 +1,244 @@
+    with respect to all differentiable inputs.
+
+    If the circuit represented by the qnode returns the expectation value of an
+    observable, it can be interpreted as a function
+    :math:`f: \mathbb{R}^N \rightarrow \mathbb{R}`. This function can always be
+    expressed by a Fourier series
+
+    .. math::
+
+        \sum \limits_{n_1\in \Omega_1} \dots \sum \limits_{n_N \in \Omega_N}
+        c_{n_1,\dots, n_N} e^{-i x_1 n_1} \dots e^{-i x_N n_N}
+
+    summing over the *frequency spectra* :math:`\Omega_i \subseteq \mathbb{Z},`
+    :math:`i=1,\dots,N`, where :math:`\mathbb{Z}` are the integers. Each
+    spectrum has the property that :math:`0 \in \Omega_i`, and the spectrum is
+    symmetric (for every :math:`n \in \Omega_i` we have that :math:`-n \in
+    \Omega_i`).
+
+    As shown in `Vidal and Theis (2019)
+    <https://arxiv.org/abs/1901.11434>`_ and `Schuld, Sweke and Meyer (2020)
+    <https://arxiv.org/abs/2008.08605>`_, if an input :math:`x_j, j = 1 \dots N`
+    only enters into single-parameter gates of the form :math:`e^{-i x_j G}`, the
+    frequency spectrum :math:`\Omega_j` is fully determined by the eigenvalues
+    of the generators :math:`G`. In many situations, the spectra are limited
+    to a few frequencies only, which in turn limits the function class that the circuit
+    can express.
+
+    Arguments to the qnode are marked as inputs :math:`x_1, \dots, x_N` by making them
+    differentiable.
+
+    .. note::
+        Differentiability of the inputs is not used here to compute
+        gradients, but to track inputs through classical
+        pre-processing and circuit compilation procedures.
+
+    Args:
+        qnode (pennylane.QNode): a quantum node that has been called with some inputs
+
+    Returns:
+        (Dict[pennylane.numpy.tensor, list[float]]): Dictionary of scalar inputs as keys, and
+        their frequency spectra as values.
+
+    **Example**
+
+    .. code-block:: python
+
+        import pennylane as qml
+        from pennylane import numpy as pnp
+        from pennylane.beta.transforms.fourier import spectrum
+
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.templates.AngleEmbedding(x[0:3], wires=[0, 1, 2])
+            qml.RX(x[0], wires=1)
+            qml.Rot(x[0], x[1], x[3], wires=1)
+            qml.CNOT(wires=[1, 2])
+            qml.RX(x[3], wires=2)
+            return qml.expval(qml.PauliZ(wires=2))
+
+        x = pnp.array([0.1, 0.2, 0.3, 0.4, 0.5], requires_grad=True)
+
+
+        res = spectrum(circuit)(x)
+
+        for inp, freqs in res.items():
+            print(f"{inp}: {freqs}")
+
+        >>> 0.1: [-3.0, -2.0, -1.0, 0.0, 1.0, 2.0, 3.0]
+        >>> 0.2: [-2.0, -1.0, 0.0, 1.0, 2.0]
+        >>> 0.3: [-1.0, 0.0, 1.0]
+        >>> 0.4: [-2.0, -1.0, 0.0, 1.0, 2.0]
+        >>> 0.5: [-1.0, 0.0, 1.0]
+
+
+    """
+
+    @wraps(qnode)
+    def wrapper(*args, **kwargs):
```

Comment:
Why is the wrapper necessary?

Reply:
I used the way other transforms do it...this allows you do have the syntax

```
qml.func(qnode)(qnode_args)
```